ostree-upload-server.py

```diff
@@ -251,16 +266,17 @@ def webapp_callback(task_name, filepath):
                         logging.debug("idle, do maintenance")
                         workers.stop()
 
-                        try:
-                            output = check_output(["flatpak",
-                                                   "build-update-repo",
-                                                   "--generate-static-deltas",
-                                                   "--prune",
-                                                   self._repo],
-                                                  stderr=STDOUT)
-                            logging.info("completed maintenance: " + output)
-                        except CalledProcessError as e:
-                            logging.error("failed maintenance: " + e.output)
+                        with RepoLock(self._repo, exclusive=True):
+                            try:
+                                output = check_output(["flatpak",
+                                                       "build-update-repo",
+                                                       "--generate-static-deltas",
+                                                       "--prune",
+                                                       self._repo],
+                                                      stderr=STDOUT)
+                                logging.info("completed maintenance: " + output)
+                            except CalledProcessError as e:
+                                logging.error("failed maintenance: " + e.output)
```

Comment:
I think this doesn't print everything about the error. I'd go with the following since loggers usually know how to handle exceptions properly:
```
logging.error("ERROR! Maintenance task failed!")
logging.error(e)
```